Mathlib/RingTheory/TensorProduct/Nontrivial.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.LinearAlgebra.Basis.VectorSpace
+import Mathlib.RingTheory.Flat.Basic
+import Mathlib.RingTheory.Localization.FractionRing
+
+/-!
+
+# Nontriviality of tensor product of algebras
+
+This file contains some more results on nontriviality of tensor product of algebras.
+
+-/
+
+open TensorProduct
+
+namespace Algebra.TensorProduct
+
+/-- If `A`, `B` are nontrivial algebras over a field `F`, then `A ⊗[F] B` is nontrivial. -/
+theorem nontrivial_of_field
+    (F A B : Type*) [Field F] [CommRing A] [CommRing B] [Algebra F A] [Algebra F B]
+    [Nontrivial A] [Nontrivial B] : Nontrivial (A ⊗[F] B) :=
+  nontrivial_of_algebraMap_injective_of_flat_left F A B (RingHom.injective _)
```

Comment:
This should be `by infer_instance` if you import `Mathlib.RingTheory.Flat.FaithfullyFlat`.

Reply:
Thanks! I've already noticed that result. But I'm not aware that it is an `instance`. I'll refactor my code.

Reply:
> Were you tripped by [leanprover/doc-gen4#240](https://github.com/leanprover/doc-gen4/issues/240)?

Yes 🤣

---

Let me evaluate this suggestion. I don't know how heavy is `Mathlib.RingTheory.Flat.FaithfullyFlat`.

Reply:
I think we should treat `Flat.FaithfullyFlat` as a basic file. In particular, your file is called `TensorProduct/Nontrivial.lean`. My first instinct would be that it will involve some faithfully flat argument.